File: fibonazzi/fibonazzi.js
function fibonazzi(n) {
	if (n === 0) return 0;
	if (n === 1) return 1;

	const current = fibonazzi(n - 1);
	const next = fibonazzi(n - 2);

	return current + next;
}

console.log(fibonazzi(0)); // 0
console.log(fibonazzi(1)); // 1
console.log(fibonazzi(2)); // 1
console.log(fibonazzi(3)); // 2
console.log(fibonazzi(4)); // 3
console.log(fibonazzi(5)); // 5
console.log(fibonazzi(8)); // 8

function memoizedFibonazzi(n, memo = {}) {
	if (n in memo) return memo[n];

	if (n === 0) return 0;
	if (n === 1) return 1;

	memo[n] = memoizedFibonazzi(n - 1, memo) + memoizedFibonazzi(n - 2, memo);

	return memo[n];
}

console.log(memoizedFibonazzi(0)); // 0
console.log(memoizedFibonazzi(1)); // 1
console.log(memoizedFibonazzi(2)); // 1
console.log(memoizedFibonazzi(3)); // 2
console.log(memoizedFibonazzi(4)); // 3
console.log(memoizedFibonazzi(5)); // 5
console.log(memoizedFibonazzi(8)); // 8

File: fibonazzi/fibonazzi.py
def fibonazzi(n):
	if n == 0: return 0
	if n == 1: return 1

	return fibonazzi(n - 1) + fibonazzi(n - 2)

print(fibonazzi(0))
print(fibonazzi(1))
print(fibonazzi(2))
print(fibonazzi(3))
print(fibonazzi(4))
print(fibonazzi(5))
print(fibonazzi(8))

def memoized_fibonazzi(n, memo = {}):
	if n in memo: return memo[n]

	if n == 0: return 0
	if n == 1: return 1

	memo[n] = memoized_fibonazzi(n - 1, memo) + memoized_fibonazzi(n - 2, memo)

	return memo[n]

print("---- Memoized fibonazzi ----")
print(memoized_fibonazzi(0))
print(memoized_fibonazzi(1))
print(memoized_fibonazzi(2))
print(memoized_fibonazzi(3))
print(memoized_fibonazzi(4))
print(memoized_fibonazzi(5))
print(memoized_fibonazzi(8))
